tests/kubernetes-agent/KubernetesAgent.IntegrationTests/RunsAgentUpgrade.cs

```diff
@@ -0,0 +1,89 @@
+using Halibut;
+using KubernetesAgent.Integration.Setup;
+using KubernetesAgent.Integration.Setup.Common;
+using Octopus.Tentacle.Client;
+using Octopus.Tentacle.Client.Scripts.Models;
+using Octopus.Tentacle.Client.Scripts.Models.Builders;
+using Octopus.Tentacle.Contracts;
+using Xunit.Abstractions;
+
+namespace KubernetesAgent.Integration;
+
+public class HelmUpgradeTests(ITestOutputHelper output) : IAsyncLifetime
+{
+    readonly ITestOutputHelper output = output;
+    ILogger logger = null!;
+    readonly TemporaryDirectory workingDirectory = new(Directory.CreateTempSubdirectory());
+    KubernetesClusterInstaller clusterInstaller = null!;
+    KubernetesAgentInstaller agentInstaller  = null!;
+    TentacleClient client = null!;
+    string kindExePath = null!;
+    string helmExePath = null!;
+    string kubeCtlPath = null!;
+    
+    public async Task InitializeAsync()
+    {
+        logger = new LoggerConfiguration()
+            .MinimumLevel.Verbose()
+            .WriteTo.TestOutput(output)
+            .CreateLogger();
+
+        var requiredToolDownloader = new RequiredToolDownloader(workingDirectory, logger);
+        (kindExePath, helmExePath, kubeCtlPath) = await requiredToolDownloader.DownloadRequiredTools(CancellationToken.None);
+        clusterInstaller =  new KubernetesClusterInstaller(workingDirectory, kindExePath, helmExePath, kubeCtlPath, logger);
+        await clusterInstaller.Install();
+        agentInstaller = new KubernetesAgentInstaller(workingDirectory , helmExePath, kubeCtlPath, clusterInstaller.KubeConfigPath, logger);
+        client = await agentInstaller.InstallAgent();
+    }
+
+    public async Task DisposeAsync()
+    {
+        clusterInstaller.Dispose();
+        workingDirectory.Dispose();
+        await Task.CompletedTask;
+    }
+    
+    [Fact]
+    public async Task CanUpgradeAgentAndRunCommand()
+    {
+        var helmPackage = HelmChartBuilder.BuildHelmChart(helmExePath, workingDirectory);
+        var helmPackageFile = new FileInfo(helmPackage);
+        var packageName = helmPackageFile.Name;
+        var packageBytes = await File.ReadAllBytesAsync(helmPackage);
+        var helmUpgradeScript =
+            $"""
+             helm upgrade \
+             --atomic \
+             --namespace {agentInstaller.Namespace} \
+             {agentInstaller.AgentName} \
+             /tmp/{packageName}
+             """;
+        var scriptCommand = new ExecuteKubernetesScriptCommandBuilder(Guid.NewGuid().ToString())
+            .WithScriptBody($"cp ./{packageName} /tmp/{packageName} && {helmUpgradeScript}")
```

Comment:
The reason I haven't is because the copy and upgrade need to happen in the same execution context, since `/tmp` won't make it into the next script pod!

Reply:
ah of course 😄